### ISSUE TITLE
fix: divergence in maskl output for kind = 8

### ISSUE THREAD

**Issue by HarshitaKalani** [lfortran/lfortran#4242]
``fortran
program main
integer, dimension(2,2) :: x
x = 2
print*, maskl(x, 8)
end program 
```
```console
$ gfortran a.f90 && ./a.out
 -4611686018427387904 -4611686018427387904 -4611686018427387904 -4611686018427387904
(lf) harshita@harshita-linux:~/Desktop/lfortran$ lfortran a.f90
-1073741824 -1073741824 
-1073741824 -1073741824 

```

**Comment by kmr-srbh** [comment]
Please see https://github.com/lfortran/lfortran/issues/4272#issuecomment-2171744169.

**Comment by Pranavchiku** [comment]
I do not think it has something to do with https://github.com/lfortran/lfortran/issues/4272#issuecomment-2171744169 the original problem is in implementation whereas the comment feels wrong to me. When you see implementation of `maskl` it has:

```cpp
        /*
        * r = Maskl(x)
        * r = (x == 64) ? -1 : ((1 << x) - 1) << (64 - x)
        */
```

And I think `x == 64` handles only `kind=8` correctly. So that is wrong, it has nothing to do with `return_types` and stuff.

cc @HarshitaKalani @kmr-srbh

**Comment by HarshitaKalani** [comment]
Exactly @Pranavchiku . I agree with you. We need to add support for kind = 4 correctly in eval_maskl .

**Comment by HarshilShah1804** [comment]
@HarshitaKalani This is now working fine in lfortran. I think we can close this issue.

**Comment by HarshitaKalani** [comment]
Yes. it indeed works. @HarshilShah1804 can you please add a test for this? We can then close it.
